File: www/js/rpc_utils.js
var session = new SessionStorage();
session.createItem("session_id", "");

function rpc_call(arg)
{
    var rqst_json = JSON.stringify({session: session.getItem('session_id'), call: arg});
    var ret = null;

    $.ajax({url: "http://" + document.location.host + "/rpc", type: "post", data: {request: rqst_json}, async: false, success: function(data)
    {
        ret = data;
        //alert(ret);
    }});

    if(ret.session != undefined)
    {
        session.setItem('session_id', ret.session)
    }

    //ret = jQuery.parseJSON(ret);
    if(ret.rpc_status == false)
    {
        alert("RPC Error:\n" + ret.rpc_fail_msg)
    }
    else
    {
        return ret.return_value;
    }
}

function filenameFromPath(path)
{
    var idx = path.lastIndexOf('/');
    return path.substring(idx + 1);
}

File: rpc_utils.py
import json

import cherrypy

def rpc_enable(session_check=True, login_check=True):
    def wrapper(f):
        f.rpc_enable = True
        f.rpc_session_check = session_check
        f.rpc_login_check = login_check
        return f

    return wrapper


def rpc_return(success, data, session=None):
    if session == None:
        return json.dumps({'rpc_status': True, 'return_value': {'success': success, 'data': data}})
    else:
        return json.dumps({'rpc_status': True, 'session': session, 'return_value': {'success': success, 'data': data}})
